Add reference framework image upload feature

samshara opened the pull request:
- fixes https://github.com/the-deep/client/issues/1672
- depends on https://github.com/the-deep/client/pull/1777

## Changes
 * Add reference framework image upload feature

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] permission checks
- [x] translations

samshara committed:
Add reference framework image upload feature

tnagorra commented on src/newViews/AnalyticalFramework/FrameworkDetails/FrameworkDetailsForm/index.tsx:
```diff
@@ -58,6 +65,11 @@ function FrameworkDetailsForm(props: Props) {
         setOrganizationOptions,
     ] = useState<BasicOrganization[] | undefined | null>();
 
+    const analyticalFramework: PartialFormType = pick(
+        analyticalFrameworkFromProps ?? defaultFormValues,
+        ['title', 'organization', 'description', 'isPrivate', 'createdByName', 'createdAt'],
+    );
+
```
I don't think we need this.

tnagorra committed:
Send empty string when the value is undefined for form data

- Do not send content type when sending form data

tnagorra commented on src/newViews/AnalyticalFramework/FrameworkDetails/FrameworkDetailsForm/UploadImage/index.tsx:
```diff
@@ -0,0 +1,75 @@
+    className?: string;
+    src?: string;
+    alt: string;
+    name: N;
```
Let's make this a controlled component and pass a value through.

samshara committed:
Pass value to UploadImage component

tnagorra committed:
Fix typings

- Fix DeepFileInput behavior on change
- UploadImage to accept a image using different prop
- Fix parent af update after form submit
